nucleus/deploy/client.py

```diff
@@ -28,28 +28,53 @@
 class DeployClient:
     """Scale Deploy Python Client extension."""
 
-    def __init__(self, api_key: str, endpoint: str = SCALE_DEPLOY_ENDPOINT):
+    def __init__(
+        self,
+        api_key: str,
+        endpoint: str = SCALE_DEPLOY_ENDPOINT,
+        is_internal: bool = False,
+    ):
         """
         Initializes a Scale Deploy Client.
 
         Parameters:
             api_key: Your Scale API key
             endpoint: The Scale Deploy Endpoint (this should not need to be changed)
+            is_internal: True iff you are connecting to a self-hosted Scale Deploy
         """
         self.connection = Connection(api_key, endpoint)
+        self.is_internal = is_internal
+        self.upload_bundle_fn: Optional[Callable[[str, str], None]] = None
 
     def __repr__(self):
         return f"DeployClient(connection='{self.connection}')"
 
     def __eq__(self, other):
         return self.connection == other.connection
 
+    def register_upload_bundle_fn(
```

Comment:
Not sure if this is the best way of allowing users to use their own backend for storing model bundles (in internal mode).